### PR TITLE
runfix: Warn user when applock is disabled on the team level

### PR DESCRIPTION
**Pull request by atomrc**
## Description

Will fix the flow of disabling appLock for an entire team. In this case, individual users should be warned that the lock has been removed. 

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/page/components/FeatureConfigChange/FeatureConfigChangeNotifier/FeatureConfigChangeNotifier.ts [hunk 1]
```diff
@@ -44,15 +43,14 @@ import {loadFeatureConfig, saveFeatureConfig} from './FeatureConfigChangeNotifie
...
-      newConfig?: FeatureWithoutConfig | Feature<any> | undefined,
-    ) => undefined | {htmlMessage: string; title: StringIdentifer; primaryAction?: Action}
-  >
-> = {
+type FeatureMessageGenerator = {
+  [K in keyof FeatureList]: (
+    oldConfig?: FeatureList[K],
+    newConfig?: FeatureList[K],
+  ) => undefined | {htmlMessage: string; title: StringIdentifer; primaryAction?: Action};
+};
```
Comment (atomrc) [inline src/script/page/components/FeatureConfigChange/FeatureConfigChangeNotifier/FeatureConfigChangeNotifier.ts:51]: This is an improvement of the type, now the `oldConfig` and `newConfig` will be correctly typed with the actual feature config

**Review by PatrykBuniX — APPROVED** [review]
Looks good 🚀

File: src/script/page/components/FeatureConfigChange/FeatureConfigChangeNotifier/FeatureConfigChangeNotifier.ts [hunk 2]
```diff
@@ -79,6 +77,18 @@ const featureNotifications: Partial<
       title: 'featureConfigChangeModalAudioVideoHeadline',
     };
   },
+
+  [FEATURE_KEY.APPLOCK]: (oldConfig, newConfig) => {
+    const shouldWarn = oldConfig?.config.enforceAppLock === true && newConfig?.config.enforceAppLock === false;
```
Comment (przemvs) [inline src/script/page/components/FeatureConfigChange/FeatureConfigChangeNotifier/FeatureConfigChangeNotifier.ts:82]: WDYT? :)

```suggestion
    const shouldWarn = !!oldConfig?.config.enforceAppLock && !newConfig?.config.enforceAppLock;
```

Comment (atomrc) [reply src/script/page/components/FeatureConfigChange/FeatureConfigChangeNotifier/FeatureConfigChangeNotifier.ts:82]: That would not work, we want to check that they really are `true` or `false`. `undefined` should be excluded

**Review by przemvs — APPROVED** [review]

**Pull request merged**